Flakey test "TestNodeTraitRegression.test_wright_fisher_unsimplified"

I've seen this test fail on CI at a rate of a couple a week.
```
 =================================== FAILURES ===================================
 ___________ TestNodeTraitRegression.test_wright_fisher_unsimplified ____________
 [gw1] darwin -- Python 3.6.11 /usr/local/miniconda/envs/anaconda-client-env/bin/python3.6
 
 self = <tests.test_tree_stats.TestNodeTraitRegression object at 0x7fe6c4bda048>
 
     def test_wright_fisher_unsimplified(self):
         tables = wf.wf_sim(
             4,
             5,
             seed=1,
             deep_history=True,
             initial_generation_samples=False,
             num_loci=5,
         )
         tables.sort()
         ts = tables.tree_sequence()
 >       self.verify(ts)
 
 tests/test_tree_stats.py:594: 
 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
 tests/test_tree_stats.py:4488: in verify
     self.verify_trait_regression(ts, W, Z, windows=windows)
 tests/test_tree_stats.py:4546: in verify_trait_regression
     self.assertArrayAlmostEqual(sigma1, sigma3)
 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
 
 self = <tests.test_tree_stats.TestNodeTraitRegression object at 0x7fe6c4bda048>
 x = array([[[0.        ],
         [0.        ],
         [0.        ],
         [0.        ],
         [0.        ],
         ...65014],
         [0.        ],
         [0.2617744 ],
         [0.        ],
         [0.        ],
         [0.19294995]]])
 y = array([[[0.],
         [0.],
         [0.],
         [0.],
         [0.],
         [0.],
         [0.],
         [0.],
       ...[0.],
         [0.],
         [0.],
         [0.],
         [0.],
         [0.],
         [0.],
         [0.],
         [0.]]])
 atol = 1e-06, rtol = 1e-07
 
     def assertArrayAlmostEqual(self, x, y, atol=1e-6, rtol=1e-7):
 >       nt.assert_allclose(x, y, atol=atol, rtol=rtol)
 E       AssertionError: 
 E       Not equal to tolerance rtol=1e-07, atol=1e-06
 E       
 E       Mismatched elements: 103 / 315 (32.7%)
 E       Max absolute difference: 0.26177442
 E       Max relative difference: inf
 E        x: array([[[0.      ],
 E               [0.      ],
 E               [0.      ],...
 E        y: array([[[0.],
 E               [0.],
 E               [0.],...
 
 tests/test_tree_stats.py:541: AssertionError
```

--- a/python/tests/test_tree_stats.py
+++ b/python/tests/test_tree_stats.py
@@ -930,6 +930,7 @@ class WeightStatsMixin:
         """
         Generate a series of example weights from the specfied tree sequence.
         """
+        np.random.seed(46)
         for k in [min_size, min_size + 1, min_size + 10]:
             W = 1.0 + np.zeros((ts.num_samples, k))
             W[0, :] = 2.0
@@ -3377,6 +3378,7 @@ class TestSampleSets(StatsTestCase):
             ts.sample_count_stat([[ts.num_samples]], self.identity_f(ts), 1)
 
     def test_span_normalise(self):
+        np.random.seed(92)
         ts = self.get_example_ts()
         sample_sets = [[0, 1], [2, 3, 4], [5, 6]]
         windows = ts.sequence_length * np.random.uniform(size=10)
@@ -4323,7 +4325,11 @@ def regression(y, x, z):
         if np.linalg.matrix_rank(xz) == xz.shape[1]:
             z = xz
     xz = np.column_stack([x, z])
-    if np.linalg.matrix_rank(xz) < xz.shape[1]:
+    # check if y is sufficiently independent of the subspace spanned by xz
+    Pz = np.matmul(z, np.linalg.pinv(z))
+    Py = np.matmul(Pz, y)
+    denom = np.sum((y - Py) ** 2)
+    if np.linalg.matrix_rank(xz) < xz.shape[1] or denom < 1e-8:
         return 0.0
     else:
         coefs, _, _, _ = np.linalg.lstsq(xz, y, rcond=None)
@@ -4455,6 +4461,7 @@ class TestTraitRegression(StatsTestCase, WeightStatsMixin):
         return ts
 
     def example_covariates(self, ts):
+        np.random.seed(999)
         N = ts.num_samples
         for k in [1, 2, 5]:
             k = min(k, ts.num_samples)

--- a/python/tskit/trees.py
+++ b/python/tskit/trees.py
@@ -5675,7 +5675,7 @@ class TreeSequence:
     ):
         """
         For each trait w (i.e., each column of W), performs the least-squares
-        linear regression :math:`w ~ g + Z`,
+        linear regression :math:`w \\sim g + Z`,
         where :math:`g` is inheritance in the tree sequence and the columns of :math:`Z`
         are covariates, and computes the squared coefficient of :math:`g` in this
         regression.

